opentracing/src/main/java/opentracing/TraceCodec.java

```diff
@@ -24,16 +24,16 @@
  * binary message format, so that it can be inspected efficiently by the middleware / routing layers
  * without parsing the whole message.
  */
-public interface TraceContextCodec {
+public interface TraceCodec {
```

Comment:
@yurishkuro does a good job of demonstrating what's the intended use of the codec in https://github.com/opentracing/opentracing.github.io/pull/34/files#r50616999

My misunderstanding to this has been based on my interpretation of what serialisation entails.

Serialisation is the writing of bytes or characters into streams/buffers for the sake of storage or cross-process communication.

What the API here is trying to do is transform state from one object (eg the http headers) into the tracing's SpanContext. This is a transformation process not serialisation. While http and rpc frameworks offer state structures that we'll need to transform into SpanContexts it's a leap to presume this transformation is always connected to serialisation, and worse to leak that connection to the API user.

An example of this transformation not being second-handedly connected to serialisation is when reading offline tracing data, for example from log files, where the data goes through a number of transformation/cleansing steps before being finally transformed into SpanContexts.